src/components/collection/Picker.tsx

```diff
@@ -1,85 +1,176 @@
     const [missingInput, setMissingInput] = useState(false);
 
-    const { liveSpecs: collectionData, error } = useLiveSpecs('collection');
+    const {
+        liveSpecs,
+        error: liveSpecsError,
+        isValidating: isValidatingLiveSpecs,
+    } = useLiveSpecs('collection');
+
+    // Draft Editor Store
+    const persistedDraftId = useEditorStore_persistedDraftId();
 
     // Form State Store
-    const messagePrefix = useFormStateStore_messagePrefix();
+    const formActive = useFormStateStore_isActive();
 
     // Resource Config Store
     const collections = useResourceConfig_collections();
+    const discoveredCollections = useResourceConfig_discoveredCollections();
+    const setRestrictedDiscoveredCollections =
+        useResourceConfig_setRestrictedDiscoveredCollections();
 
     const setResourceConfig = useResourceConfig_setResourceConfig();
 
+    const {
+        draftSpecs,
+        error: draftSpecsError,
+        isValidating: isValidatingDraftSpecs,
+    } = useDraftSpecs(persistedDraftId);
+
+    useEffect(() => {
+        const populateCollectionData =
+            entityType === ENTITY.MATERIALIZATION
+                ? liveSpecs.length > 0
+                : !isValidatingLiveSpecs && !isValidatingDraftSpecs;
+
+        if (populateCollectionData) {
+            let collectionsOnServer: CollectionData[] = liveSpecs.map(
+                ({ catalog_name }) => ({
+                    name: catalog_name,
+                    classification: 'Existing Collections',
+                })
+            );
+
+            if (entityType === ENTITY.CAPTURE) {
+                collectionsOnServer = [
+                    ...draftSpecs
+                        .filter(
+                            ({ spec_type }) => spec_type === ENTITY.COLLECTION
+                        )
+                        .map(({ catalog_name }) => ({
+                            name: catalog_name,
+                            classification: 'Discovered Collections',
+                        })),
+                    ...collectionsOnServer,
+                ];
+            }
+
+            setCollectionData(collectionsOnServer);
+        }
+    }, [
+        setCollectionData,
+        collections,
+        draftSpecs,
+        entityType,
+        isValidatingDraftSpecs,
+        isValidatingLiveSpecs,
+        liveSpecs,
+        workflow,
+    ]);
+
     const handlers = {
-        updateCollections: (event: React.SyntheticEvent, value: any) => {
-            setResourceConfig(value);
+        updateCollections: (
+            _event: React.SyntheticEvent,
+            value: CollectionData[]
+        ) => {
+            setResourceConfig(value.map(({ name }) => name));
+
+            value
+                .filter(({ name }) => discoveredCollections?.includes(name))
+                .forEach(({ name }) => {
+                    setRestrictedDiscoveredCollections(name);
```

Comment:
Remove a given, discovered collection from the list of restricted discovered collections when reselected.